internal/compiler-interface/src/main/java/xsbti/BasicVirtualFileRef.java

```diff
@@ -33,6 +36,15 @@ public String name() {
         return name;
     }
 
+    public String[] names() {
+        if (parent == "") {
+            return new String[] { name };
+        }
+        ArrayList<String> xs = new ArrayList<String>(Arrays.asList(parent.split("/")));
```

Comment:
Not a big deal at all, but I kinda prefer an imperative approach here just to minimize the extra copying (it also isn't obvious from the code alone that you need to wrap the result of Arrays.asList because otherwise `xs.add` will throw an unsupported operation exception). Maybe that's a premature optimization though:
```java
String[] parts = parent.split("/");
String[] results = new String[parts.length + 1];
int i = 0;
for (i = 0; i < parts.length; i++) results[i] = parts[i];
results[i] = name;
return results;  

```